spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/autoconfig/TraceAutoConfiguration.java

```diff
@@ -62,10 +69,24 @@ public Sampler defaultTraceSampler() {
 	@Bean
 	@ConditionalOnMissingBean(Tracer.class)
 	public DefaultTracer traceManager(Sampler sampler, Random random,
-									SpanNamer spanNamer, SpanLogger spanLogger,
-									SpanReporter spanReporter) {
+			SpanNamer spanNamer, SpanLogger spanLogger,
+			SpanReporter spanReporter, List<SpanJoiner> spanJoiners,
+			List<SpanInjector> spanInjectors) {
 		return new DefaultTracer(sampler, random, spanNamer, spanLogger,
-				spanReporter);
+				spanReporter, new SpanJoinerComposite(spanJoiners),
+				new SpanInjectorComposite(spanInjectors));
+	}
+
+	// have to register a NoOp - otherwise autowiring won't work for trace manager
```

Comment:
I've already tried that and it won't work cause Tracer is also a `SpanJoiner` so you will have a circular dependency cycle.